motionplan/plannerOptions.go

```diff
@@ -198,18 +206,100 @@ type plannerOptions struct {
 
 	PlanningAlgorithmSettings AlgorithmSettings `json:"planning_algorithm_settings"`
 
-	Fallback *plannerOptions `json:"fallback_options"`
+	RandomSeed int `json:"rseed"`
+
+	PathStepSize float64 `json:"path_step_size"`
+
+	MeshesAsOctrees bool `json:"meshes_as_octrees"`
+
+	Fallback *PlannerOptions `json:"fallback_options"`
 
 	TimeMultipleAfterFindingFirstSolution int `json:"time_multiple_after_finding_first_solution"`
 }
 
+// NewPlannerOptionsFromExtra returns basic default settings updated by overridden parameters
+// found in the "extra" of protobuf MoveRequest. The extra must be converted to an instance of
+// map[string]interface{} first.
+func NewPlannerOptionsFromExtra(extra map[string]interface{}) (*PlannerOptions, error) {
+	opt := NewBasicPlannerOptions()
+
+	jsonString, err := json.Marshal(extra)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(jsonString, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	if opt.CollisionBufferMM < 0 {
+		return nil, errors.New("collision_buffer_mm can't be negative")
+	}
+
+	// we want to deprecate, rather than break, usage of the "tolerance" key for
+	// OrientationMotionProfile
+	if opt.MotionProfile == OrientationMotionProfile {
+		opt.OrientationTolerance = opt.ToleranceFactor
+	}
+	return opt, nil
+}
+
+// Returns an updated PlannerOptions taking into account whether TP-Space is being used and whether
+// a Free or Position-Only motion profile was requested with an unspecified algorithm (indicating the desire
+// to let motionplan handle what algorithm to use and allow for a fallback).
+func updateOptionsForPlanning(opt *PlannerOptions, useTPSpace bool) (*PlannerOptions, error) {
```

Comment:
This whole function seems kinda fishy from an API perspective.  I would rather not be mutating options like this.  And the options themself already have some knowledge of the motion profile and TPspace usage so I don't see why its necessary.  Totally get it if this is just the easiest way to do a 1:1 change though